### PR TITLE
integration-cli: Fix path problems in TestBuildRenamedDockerfile

### PR DESCRIPTION
**Pull request by ahmetb**
`TestBuildRenamedDockerfile` tests hard-code unix-style
path building. Made use of `path/filepath` to make these
tests work on Windows as well.

Signed-off-by: Ahmet Alp Balkan ahmetalpbalkan@gmail.com
cc: @duglin @unclejack @jfrazelle @icecrime @tiborvass

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
> https://jenkins.dockerproject.com/job/Windows-PRs/55/console
=== RUN TestBuildRenamedDockerfile
[PASSED]: build - rename dockerfile
--- PASS: TestBuildRenamedDockerfile (2.75s) :+1:

File: integration-cli/docker_cli_build_test.go [hunk 1]
```diff
@@ -4583,17 +4583,21 @@ func TestBuildRenamedDockerfile(t *testing.T) {
 		t.Fatalf("Should have used dFile, output:%s", out)
 	}
 
-	out, _, err = dockerCmdInDir(t, ctx.Dir, "build", "--file=/etc/passwd", "-t", "test5", ".")
+	dirWithNoDockerfile, err := ioutil.TempDir(os.TempDir(), "empty")
+	if err != nil {
+		t.Fatal(err)
+	}
+	out, _, err = dockerCmdInDir(t, ctx.Dir, "build", fmt.Sprintf("--file=%s", dirWithNoDockerfile), "-t", "test5", ".")
```
Comment (duglin) [inline integration-cli/docker_cli_build_test.go]: This one is supposed to point to an existing file so I'm not sure this is the same testcase.
We may need to create a temp dir+file and then point to that?

**Comment by duglin** [comment]
You may want to wait for https://github.com/docker/docker/pull/10555 to get merged first.

File: integration-cli/docker_cli_build_test.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline integration-cli/docker_cli_build_test.go]: @duglin ah, makes sense. For some reason I guess I thought /etc/passwd as a dir. :-/ I'll fix this.

**Comment by ahmetalpbalkan** [comment]
@duglin makes sense. please get it merged :P

**Comment by duglin** [comment]
@ahmetalpbalkan I'm twisting @tiborvass 's arm as hard as I can  :-)

**Comment by jfrazelle** [comment]
ok that was merged you wanna update?

**Comment by ahmetalpbalkan** [comment]
@jfrazelle rebased on top of master and pushed.

**Comment by LK4D4** [comment]
Seems like all failed :)

**Comment by jfrazelle** [comment]
well its failing bc we need #10904 ;)

**Comment by jfrazelle** [comment]
LGTM

**Comment by jfrazelle** [comment]
ping @duglin

**Comment by ahmetalpbalkan** [comment]
> https://jenkins.dockerproject.com/job/Windows-PRs/69/console 
=== RUN TestBuildRenamedDockerfile
[PASSED]: build - rename dockerfile
--- PASS: TestBuildRenamedDockerfile (3.09s) :+1:

**Comment by duglin** [comment]
Yup - LGTM

**Pull request merged**